### DISCUSSION TITLE
`master` → `main` branch renaming

### DISCUSSION THREAD

**Discussion by sbrl** [tldr-pages/tldr#5868] [Announcements]
Calling all maintainers of both clients and tldr alike!

The `master` branch has now been renamed to `main`.

## Client authors
While a new mirror branch `master` that reflects `main` has been created, please update your clients if required to default to `main` instead of `master`. In the future the `master` mirror branch will be deleted.

Note though that it is RECOMMENDED that you download our `tldr.zip` instead of cloning this repository, as per our [client specification](https://github.com/tldr-pages/tldr/blob/main/CLIENT-SPECIFICATION.md#caching) if appropriate.

## tldr maintainers
All existing PRs and branch protection rules have been updated.

**Please watch out for new PRs that merge against `master`, as I have been unable to create a branch protection rule on `master` that doesn't also block GitHub Actions from pushing to it.**

<hr /><em>This discussion was created from the release <a href='https://github.com/tldr-pages/tldr/releases/tag/v1.5b'>`master` → `main` branch renaming</a>.</em>

**Comment by Ck-a** [comment]
Hi again been busy what is new??

**Reply by sbrl** [reply]
Hey there! I'm not sure what you're asking here?

**Comment by jxu** [comment]
This change by GitHub was IMO too breaking and has caused quite a bit of confusion for me. If I go to https://github.com/tldr-pages/tldr/tree/master, it says it is up to date with main. Yet in my local git repo, git checkout master shows only commits up to Feb 2020 and git pull or git reset does nothing. I must be misunderstanding something here.

**Reply by navarroaxel** [reply]
> This change by GitHub was IMO too breaking and has caused quite a bit of confusion for me. If I go to [`master`](https://github.com/tldr-pages/tldr/tree/master), it says it is up to date with main. Yet in my local git repo, git checkout master shows only commits up to Feb 2020.

But your issue in related to your working copy and how do you `fetch` commits from the remote.

**Reply by jxu** [reply]
Isn't the remote branch called origin/master?

**Reply by navarroaxel** [reply]
Do you get the commits when you use these commands?
```bash
git checkout master
git pull origin master -u
```

**Reply by jxu** [reply]
I think the issue is that the remote is jxu/tldr and git pull should be from upstream remote (where master branch points to to main).

**Reply by navarroaxel** [reply]
it's your responsibility keep your fork up to date. BTW actually you don't need to sync your fork, just use origin/master and push to jxu/<branch-name> to create new PRs.

**Reply by jxu** [reply]
My fork master branch _was_ up to date with git pulls, until the branch name change.

**Reply by navarroaxel** [reply]
:thinking: git pull (read) only fetch changes to your working copy, doesn't update the fork itself, you need a `push` (write) for that.

**Reply by jxu** [reply]
Ok I think I synced everything with creating the upstream remote of the official repo. (I'm pretty sure I had this before but I'm not sure what happened)

**Reply by iridacea** [reply]
> it's your responsibility keep your fork up to date. BTW actually you don't need to sync your fork, just use origin/master and push to jxu/ to create new PRs.

Just a note. That is true, but doing this way increases chances of conflict changes, and one would end up resolving that via Github's web-interface, instead of resolving it locally or even avoiding it.

**Comment by dgx-spark** [comment]
Hi there,

I am trying to install **tldr pages** with `npm` and `pip` in **WSL Kali Linux** and got this error.

```
Error: /mnt/c/Users/rsa/AppData/Roaming/npm/tldr: 11: exec: node: not found
```

![image](https://user-images.githubusercontent.com/58859932/196001168-ec9bb098-a7cd-4607-8921-2061a3ab9d69.png)

**Reply by kbdharun** [reply]
It seems that there is an error with the executable in the specified path. 

Have you tried updating or reinstalling npm? 
> I tried both the clients in Kali before it worked fine. (Not in WSL)

If npm is up to date, Try to reinstall the client by following the guide for our [npm-client](https://github.com/tldr-pages/tldr-node-client#faq).
> I would suggest you use our [npm-client](https://github.com/tldr-pages/tldr-node-client) in Kali as pip dependencies are outdated sometimes in it.

Feel free to let me know if it works

**Reply by navarroaxel** [reply]
Looks like `node` is not available in the path, are you mounting a Windows filesystem on Linux? are you using nvm?

**Comment by RoestVrijStaal** [comment]
Thanks for wasting 5 minutes of my life for this UNNEEDED change.
